docs/DeveloperGuide.md

```diff
@@ -154,6 +154,72 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+####Implementation
+`FreeScheduleCommandParser`, `FreeScheduleCommand` and `IsPersonFreePredicate` classes are involved in the execution of the `freeSchedule` command.
+
+The `parse` method inside the `FreeScheduleCommandParser` receives the user input, extracts the required arguments. It then creates a new `IsPersonFreePredicate` object that will help check if the user's contacts' schedule coincides with the specified time and date.
+
+Given below is one example usage scenario and explanation on how the `freeSchedule` feature behaves at each step. You may also refer the sequence diagram below.
+
+Step 1. The user enters `freeSchedule ti/10:00 da/2022-03-24` to find if there are any contacts who are free at the specified time and date. The arguments `ti/10:00 da/2022-03-24` are passed to the `FreeScheduleCommandParser` through its `parse` method call.
+
+Step 2. The user input `ti/10:00 da/2022-03-24` will be checked to ensure that empty input is not given. At the same time, `ParserUtil#parseTime` and `ParserUtil#parseDate` are used to check for invalid inputs.
+
+Step 3. A new `IsPersonFreePredicate` object is created and encapsulated by a new `FreeScheduleCommand` object.
+
+Step 4. The `FreeScheduleCommand` object is returned to the `LogicManager`.
+
+Step 5. During the execution of the command, the `FreeScheduleCommand` object calls `Model#updateFilteredPersonList` method with the `IsPersonFreePredicate` to update the display of the current list of contacts. If there are contacts who are free at the specified time and date, then a list of the contacts who are free will be shown. Otherwise, an empty list will be shown.
+
+Step 6. A `CommandResult` with the number of contacts free is returned. A list of contacts who are free will also be displayed to the user.
+
+####Sequence Diagram
+The following sequence diagram shows how the `freeSchedule` feature works for the example above:
+![FreeScheduleSequenceDiagram](images/FreeScheduleSequenceDiagram.png)
+
+####Activity Diagram
+The following activity diagram summarizes what happens when the `freeSchedule` feature is triggered:
+![FreeScheduleActivityDiagram](images/FreeScheduleActivityDiagram.png)
+
+####Design Considerations
+**Aspect: Should we consider the recurring events?**
+* Alternative 1: The date collision do not consider if the events will repeat in the future.
+  * Pro: 
+    * Easier implementation as we will only have to check if the events happen on the same date as the specified date.
+  * Con: 
+    * User will not be able to plan meetings with contacts in advance.
+* Alternative 2 (current implementation): The date collision considers events which will repeat in the future.
+  * Pro: 
+    * Users will be able to plan meetings with contacts in advance.
+  * Con: 
+    * Harder implementation as well will have to consider the recurring frequency every time we check the dates with the event.
+
+**Aspect: What to do with dates that have passed?**
```

Comment:
Is this part meant to explain whether we allow users to check who is free on a date that has passed? If so, I think it would be better phrased as "Should we allow users to check a date that has already passed?"

Reply:
I think I will change it to "Should we allow dates that have already passed?"

Reply:
That works too!